src/management/fs.c

```diff
@@ -430,6 +439,52 @@ static int fs_delete_process_request(struct controller_api *fs_delete,
 
 	log_debug("Deleting FS : %s.", fs_delete_api->req.fs_name);
 
+	if (fs_name_hash == NULL) {	
+		log_err("Object hash not sent");
+		rc = EINVAL;
+		request_set_errcode(request, rc);
+		fs_create_send_response(fs_delete, NULL);
+		goto error;
+	}
+
+	/* If we ever become multi-threaded, then this might not be sufficient.
+	 * Suppose we recieve 2 simultaneous request then the response to
+	 * one request should be success and the response to another request
+	 * should be "Not Found". In order to achieve this we would need a 
+	 * locking mechanism before we perform this operation. If underlying
+	 * filesystem object has its own locking mechanisms and they can ensure
+	 * that the delete operation can be performed atomically then this
+	 * won't be an issue. */
+	rc = md5hash_compute(fs_delete_api->req.fs_name, 
+			     fs_name_len, 
+			     &hash);
+	if (rc) {
+		log_err("Error Computing hash.");
```

Comment:
Need to print more debugging info here (and everywhere log_err has been added), probably a fs name, and rc.

Reply:
Yes, I would agree here since md5hash APIs internally adds log_err msgs. debugging info certainly be more helpful

Reply:
Done